src/go/rpk/pkg/config/weak.go

```diff
@@ -246,7 +246,7 @@ func (s *serverTLSArray) UnmarshalYAML(n *yaml.Node) error {
 	if err != nil {
 		return err
 	}
-	fmt.Fprintf(os.Stderr, "redpanda.yaml: %+v must be an array; a non-array value in array fields is deprecated and will be removed in the future\n", single)
+	fmt.Fprintf(os.Stderr, "redpanda.yaml: Server TLS must be an array; a non-array value in array fields is deprecated and will be removed in the future\n") // do not log serverTLS value here
```

Comment:
might want to change this comment to mention the Other field:
```
// do not log serverTLS because the Other field may contain a secret
```
perhaps above this line, because the line is already long